File: api/src/main/java/com/areeba/api/service/CustomerService.java
package com.areeba.api.service;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.data.domain.Sort;
import org.springframework.stereotype.Service;

import com.areeba.api.model.Customer;
import com.areeba.api.repository.CustomerRepository;


@Service
public class CustomerService implements CustomerServiceInterface {

	@Autowired
	private CustomerRepository customerRepository;
	
	@Override
	public Customer saveCustomer(Customer customer) {
		return customerRepository.save(customer);
	}


	@Override
	public List<Customer> getAllCustomer() {
		return customerRepository.findAll(Sort.by(Sort.Direction.DESC, "id"));
	}

	@Override
	public Customer updateCustomer(int id, Customer customer) {
		Customer customerToUpdate = customerRepository.findById(id).orElseThrow();
		customerToUpdate.setFirstName(customer.getFirstName());
		customerToUpdate.setLastName(customer.getLastName());
		customerToUpdate.setEmail(customer.getEmail());
        return customerRepository.save(customerToUpdate);
	}

	@Override
	public void deleteCustomer(int id) {
		customerRepository.deleteById(id);
	}

}

File: customer-management-frontend/src/services/CustomerService.js
import axios from "axios";

const BASE_URL = "http://localhost:8081/customer";

class CustomerService {

    //** Method to get all customers from our Api **//
    getAllCustomer(){
        return axios.get(BASE_URL);
    }
    //** Method to save customer **//
    saveCustomer(customerData){
        return axios.post(BASE_URL, customerData);
    }
    //** Method to update customer **//
    updateCustomer(id, customerData){
        return axios.put(`${BASE_URL}/${id}`, customerData)
    }
    //** Method to get customer by id **//
    getCustomerById(id){
        return axios.get(`${BASE_URL}/${id}`, id)
    }
    //** Method to delete customer **//
    deleteCustomer(id){
        return axios.delete(`${BASE_URL}/${id}`, id)
    }
}

export default new CustomerService();